geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionEventsDUnitTest.java

```diff
@@ -0,0 +1,775 @@
+
+    for (int i = 0; i < getHost(0).getVMCount(); i++) {
+      Object vmNewValue = getHost(0).getVM(i).invoke(() -> newValue);
+      assertThat(vmNewValue).as("newValue is wrong in vm " + i).isEqualTo(replaceValue);
+    }
+
+    replaceValue++;
+    int finalReplaceValue = replaceValue;
+    getHost(0).getVM(0).invoke(() -> replaceEntry(name, finalReplaceValue));
+    expectedUpdates += getHost(0).getVMCount() + 1;
+
+    assertThat(sharedCountersRule.getTotal(CREATES)).isEqualTo(expectedCreates);
+    assertThat(sharedCountersRule.getTotal(UPDATES)).isEqualTo(expectedUpdates);
+
+    assertThat(newValue).isEqualTo(replaceValue);
+
+    for (int i = 0; i < getHost(0).getVMCount(); i++) {
+      Object vmNewValue = getHost(0).getVM(i).invoke(() -> newValue);
+      assertThat(vmNewValue).as("newValue is wrong in vm " + i).isEqualTo(replaceValue);
+    }
+  }
+
+  @Test
+  public void testUpdateIsCreate() throws Exception {
+    CacheEventListener listener = new CacheEventListener() {
+
+      @Override
+      public void afterCreate(EntryEvent event) {
+        logger.info("Invoking afterCreate on listener; event={}", event);
+        sharedCountersRule.increment(CREATES);
+
+        errorCollector.checkThat(event.getDistributedMember(), equalTo(event.getCallbackArgument()));
+        errorCollector.checkThat(event.getOperation(), equalTo(Operation.CREATE));
+        if (!((EntryEventImpl) event).getInvokePRCallbacks()) {
+          errorCollector.checkThat(event.getOldValue(), nullValue());
+          errorCollector.checkThat(event.isOldValueAvailable(), equalTo(false));
+        } else {
+          errorCollector.checkThat(event.getOldValue(), nullValue());
+        }
+        errorCollector.checkThat(event.getNewValue(), equalTo(0));
+      }
+
+      @Override
+      public void afterUpdate(EntryEvent event) {
+        logger.info("Invoking afterUpdate on listener; name={}", event.getKey());
+        sharedCountersRule.increment(UPDATES);
+
+        errorCollector.checkThat(event.getOperation(), equalTo(Operation.UPDATE));
+      }
+    };
+
+    // Create in controller vm
+    createRegion(name, new SubscriptionAttributes(InterestPolicy.ALL), listener);
+
+    // Create in dunit VMs
+    for (int i = 1; i < getHost(0).getVMCount(); i++) {
+      getHost(0).getVM(i).invoke(
+          () -> createRegion(name, new SubscriptionAttributes(InterestPolicy.ALL), listener));
+    }
+
+    int expectedCreates = getHost(0).getVMCount() + 1;
+    int expectedUpdates = 0;
+
+    getHost(0).getVM(0).invoke(
+        () -> updateNewEntry(name, new SubscriptionAttributes(InterestPolicy.ALL), listener));
+
+    assertThat(sharedCountersRule.getTotal(UPDATES)).isEqualTo(expectedUpdates);
+    assertThat(sharedCountersRule.getTotal(CREATES)).isEqualTo(expectedCreates);
+  }
+
+  @Test
+  public void testObjectInvalidated() throws Exception {
+    int timeToLive = 0; // 5; // seconds expiration isn't supported in PR (yet??)
```

Comment:
we do support expiration on prs. Seems very wrong to have a test that looks like it is testing expiration invalidates but actually just calls invalidate explicitly. If this test does not need to test expiration then we should remove the expiration logic from this test.